### PR TITLE
Add version information for in-VM agent, shim, firecracker-conta…

### PR DESCRIPTION
**Pull request by kzys**
Investigating issues like #370 will be easier if the agent can report
the version.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kzys** [comment]
Right now, firecracker-containerd and firecracker-ctr have "some" version information.

```
% ./firecracker-control/cmd/containerd/firecracker-containerd -v                                    
containerd github.com/containerd/containerd 1.3.1+unknown                                           
% ./firecracker-control/cmd/containerd/firecracker-ctr -v                                           
ctr github.com/containerd/containerd 1.3.1+unknown                                                  
```

The shim is not having any.

```
% ./runtime/containerd-shim-aws-firecracker -h
Usage of ./runtime/containerd-shim-aws-firecracker:
  -address string
        grpc address back to main containerd
  -bundle string
        path to the bundle if not workdir
  -debug
        enable debug output in logs
  -httptest.serve string
        if non-empty, httptest.NewServer serves on this address and blocks
  -id string
        id of the task
  -namespace string
        namespace that owns the shim
  -publish-binary string
        path to publish binary (used for publishing events) (default "containerd")
  -socket string
        abstract socket path to serve
  -test.bench regexp
        run only benchmarks matching regexp
...
```

File: agent/main.go [hunk 1]
```diff
@@ -155,3 +167,9 @@ func main() {
 		panic(err)
 	}
 }
+
+func showVersion() {
+	// After Go 1.12, We can use runtime/debug.BuildInfo instead.
```
Comment (samuelkarp) [inline agent/main.go]: Thanks for linking this.  Seems like https://github.com/golang/go/issues/29814 is probably still a blocker though, as the main module version currently shows as "(devel)" regardless of the state of the repository from which the module was built.

Comment (kzys) [reply agent/main.go]: Oh, will add the link to the issue then.

**Comment by kzys** [comment]
Realized that we must not write anything on "start", since the stdin is used for passing the address of the shim.

* https://github.com/containerd/containerd/blob/cbfff2fb789ae8b6f2233a13fe123a1a2cd53e21/runtime/v2/README.md#start
* https://github.com/containerd/containerd/blob/324a94790d7c4d5757b89d002f7b528082fc6496/runtime/v2/shim/shim.go#L215

Will fix!

**Comment by kzys** [comment]
After the change, the shim will log the line below;

```
time="2020-01-10T20:49:24.643207368Z" level=info msg="successfully started shim (git commit: a731d81a6313ba8d78036e6bf976808554f442eb)." runtime=aws.firecracker task_id=container-TestStopVM_Isolated_ErrorExit vmID=TestStopVM_Isolated_ErrorExit
```

agent will have `-version`

```
% ./agent/agent -version
containerd Firecracker agent (git commit: a731d81a6313ba8d78036e6bf976808554f442eb)
```

firecracker-containerd follows what containerd does.

```
% ./firecracker-control/cmd/containerd/firecracker-containerd -v
containerd github.com/containerd/containerd 1.3.1+unknown a731d81a6313ba8d78036e6bf976808554f442eb
% containerd -v
containerd containerd.io 1.2.10 b34a5c8af56e510852c35414db4c1f4fa6172339
```

**Pull request merged**